common/src/test/java/io/netty/util/HashedWheelTimerTest.java

```diff
@@ -298,6 +298,30 @@ public void run(Timeout timeout) throws Exception {
         }
     }
 
+    @Test()
+    public void cancelWillCallCallback() throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final HashedWheelTimer timer = new HashedWheelTimer();
+        final Timeout t1 = timer.newTimeout(new TimerTask() {
+            @Override
+            public void run(Timeout timeout) {
+                fail();
+            }
+
+            @Override
+            public void cancelled(Timeout timeout) {
+                latch.countDown();
+            }
+        }, 90, TimeUnit.MILLISECONDS);
```

Comment:
I suspect this test could be flaky if you don't extend the timeout to something on the order of 5 seconds or more.